app/src/common/hooks/useFhirResourceTreeData.ts

```diff
@@ -196,30 +222,32 @@ const buildTree = (
     (!previousElementNode.isArray || rootNode === previousElementNode)
   ) {
     previousElementNode.children.push(currentElementNode);
-    buildTree(rest, rootNode, currentElementNode);
+    buildTree(rest, rootNode, currentElementNode, parentPath);
   } else {
     const parent = getParent(previousElementNode, rootNode);
-    if (parent) buildTree(elementsDefinition, rootNode, parent);
+    if (parent) buildTree(elementDefinitions, rootNode, parent, parentPath);
```

Comment:
if `parentPath` and `parent` are unrelated maybe we should change the naming of either ones